Automattic/wp-calypso
# Editor: Use post.utils for Status Testing

While working in `post-editor.jsx` today I noticed a few places where we are still manually testing `post.status` instead of leveraging `lib/posts/utils`.  This branch is simply some trash pickup for the locations where the post status was being manually tested.  Additionally it adds a new method for testing to see if a post's status is `private` and some tests for `isPublished()`.

__To Test__
- Open up a new post in the editor
- Privately publish a post and ensure the proper messaging is displayed in the confirmation notice, and in Ground Control
- Hard refresh the page to ensure the post was indeed published privately
- Make an update to the privately published post, verify the messaging is correct
- Create a new public post, follow the steps above to confirm messaging is correct
